jpa/subsystem/src/main/java/org/jboss/as/jpa/processor/JPADependencyProcessor.java

```diff
@@ -229,10 +217,12 @@ private int loadPersistenceUnits(final ModuleSpecification moduleSpecification,
                     // doesn't exist).
                     String providerModuleName = Configuration.getProviderModuleNameFromProviderClassName(pu.getPersistenceProviderClassName());
                     if (providerModuleName != null) {
-                        addOptionalDependency(moduleSpecification, moduleLoader, deploymentUnit, ModuleIdentifier.fromString(providerModuleName));
-                        ROOT_LOGGER.debugf("%s is configured to use persistence provider '%s', adding an optional dependency on module '%s'",
-                                pu.getPersistenceUnitName(), pu.getPersistenceProviderClassName(), providerModuleName);
+                        String moduleIdentifier = providerModuleName;
+                        moduleSpecification.addSystemDependency(new ModuleDependency(moduleLoader, moduleIdentifier, true, false, false, false));
+                        ROOT_LOGGER.debugf("added %s dependency to %s", moduleIdentifier, deploymentUnit.getName());
                     }
+                    ROOT_LOGGER.debugf("%s is configured to use persistence provider '%s', adding an optional dependency on module '%s'",
+                            pu.getPersistenceUnitName(), pu.getPersistenceProviderClassName(), providerModuleName);
```

Comment:
This probably shouldn't be displayed if `providerModuleName` is `null`?

Reply:
Addressed ^